lib/automate_liveness_agent/main.rb

```diff
@@ -86,16 +89,22 @@ def load_config
     end
 
     def set_privileges
+      unless config.scheduled_task_mode
+        @pidfile_handle = File.open(PIDFILE_LOCATION, WRITE_MODE, 0644)
```

Comment:
This means that if we fail after this point, we'll leave an empty pidfile around. We can potentially clean that up as long as we fail before we drop root privileges, but after that we can't.